### PR TITLE
Graphics: Intoducing the LIghting Buffer where to accumulate the lighted information

### PR DESCRIPTION
**Pull request by samcake**
- Introduce the LIghting Buffer and Framebuffer where to accumulate the lighting passes from deferred and transparent (to be done)

- Made a custom pixel shader to do the current copyBack pass of the deferred effect to gamma correct the lighted pixels back into the "final" color buffer

- Removed the unnecessary blending which was happening during this step

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 72d393905b36cbf65c7686ee67210c3beaaa1fab available at: https://gist.github.com/03acf77974020174c2b6<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6545.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6545.dmg

File: libraries/render-utils/src/FramebufferCache.cpp [hunk 1]
```diff
@@ -74,6 +76,12 @@ void FramebufferCache::createPrimaryFramebuffer() {
     _selfieFramebuffer = gpu::FramebufferPointer(gpu::Framebuffer::create());
     auto tex = gpu::TexturePointer(gpu::Texture::create2D(colorFormat, width * 0.5, height * 0.5, defaultSampler));
     _selfieFramebuffer->setRenderBuffer(0, tex);
+
+    _lightingTexture = gpu::TexturePointer(gpu::Texture::create2D(gpu::Element(gpu::VEC4, gpu::NUINT8, gpu::RGBA), width, height, defaultSampler));
+        //_lightingTexture = gpu::TexturePointer(gpu::Texture::create2D(gpu::Element(gpu::VEC4, gpu::NUINT8, gpu::SRGBA), width, height, defaultSampler));
+        // _lightingTexture = gpu::TexturePointer(gpu::Texture::create2D(gpu::Element(gpu::VEC4, gpu::HALF, gpu::RGBA), width, height, defaultSampler));
```
Comment (Atlante45) [inline libraries/render-utils/src/FramebufferCache.cpp]: Dead code?

File: libraries/render-utils/src/DeferredLightingEffect.cpp [hunk 2]
```diff
@@ -143,12 +143,42 @@ void DeferredLightingEffect::init(AbstractViewStateInterface* viewState) {
...
+        }
+        
+        )SCRIBE";
+        auto blitPS = gpu::ShaderPointer(gpu::Shader::createPixel(std::string(BlitTextureGamma_frag)));
+ 
+        //auto blitProgram = gpu::StandardShaderLib::getProgram(gpu::StandardShaderLib::getDrawViewportQuadTransformTexcoordVS, gpu::StandardShaderLib::getDrawTexturePS);
+        auto blitVS = gpu::StandardShaderLib::getDrawViewportQuadTransformTexcoordVS();
+        auto blitProgram = gpu::ShaderPointer(gpu::Shader::createProgram(blitVS, blitPS));
+        
+        //auto blitProgram = gpu::StandardShaderLib::getProgram(gpu::StandardShaderLib::getDrawViewportQuadTransformTexcoordVS, gpu::StandardShaderLib::getDrawTexturePS);
```
Comment (Atlante45) [inline libraries/render-utils/src/DeferredLightingEffect.cpp]: Dead code?

File: libraries/render-utils/src/DeferredLightingEffect.cpp [hunk 3]
```diff
@@ -143,12 +143,42 @@ void DeferredLightingEffect::init(AbstractViewStateInterface* viewState) {
...
+        
+        void main(void) {
+            outFragColor = texture(colorMap, varTexCoord0);
+            // if (gl_FragCoord.x > 1000) {
+            // Manually gamma correct from Ligthing BUffer to color buffer
+            outFragColor.xyz = pow( outFragColor.xyz , vec3(1.0 / 2.2) );
+            // }
+        }
+        
+        )SCRIBE";
```
Comment (Atlante45) [inline libraries/render-utils/src/DeferredLightingEffect.cpp]: Why not have that in a .slf?

**Pull request merged**